torch/csrc/lazy/core/lazy_mode.cpp

```diff
@@ -0,0 +1,75 @@
+#include <torch/csrc/lazy/core/lazy_mode.h>
+
+#include <c10/core/DispatchKey.h>
+#include <c10/core/impl/LocalDispatchKeySet.h>
+#include <torch/csrc/lazy/backend/backend_device.h>
+#include <torch/csrc/lazy/core/lazy_graph_executor.h>
+#include <torch/library.h>
+
+// TODO(whc) we can't actually depend on ts backend code from here, but we could refactor if reusing the callback
+// turns out to be the best way to implement this
+#include <torch/csrc/lazy/ts_backend/ts_eager_fallback.h>
+
+namespace torch {
+namespace lazy {
+
+std::atomic<bool>& in_lazy_mode() {
+    static std::atomic<bool> _in_lazy_mode{false};
```

Comment:
why atomic over thread-local? 

The Dispatcher's local dispatch keyset is thread-local so some threads could have Lazy Mode on and some could have it off